### ISSUE TITLE
Downloading maps of all regions of Norway doesn't download Oslo

### ISSUE THREAD

**Issue by dafadllyn** [osmandapp/OsmAnd#20127]
### Description

Downloading the maps of all regions of Norway doesn't download the map of Oslo.

### Steps to reproduce

1. Touch on "Maps & Resources" in the hamburger menu.
2. Under "Regions" select "Europe" and then "Norway".
3. Under "Regionwide maps" touch on the download button to the right of "Standard map".
4. Make sure that all regions are selected. (Note that OsmAnd only lists 12 regions although one screen back under "Regions" it lists 13.)
5. Touch on "Download".

### Actual result

The maps of all regions except of Oslo is downloaded.

### Expected result

The map of Oslo should also have been downloaded.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: 4.7.17
Android/iOS version: Android 14
Device model: Google Pixel 6a
Crash-Logs: ?
```

**Comment by scaidermern** [comment]
> ### Expected result
> 
> The map of Oslo should also have been downloaded.

Why?

The map for region Viken already contains Oslo. If you download the map for Oslo while having the map for Viken downloaded you will have duplicated data, leading to various problems (#15039).

**Comment by yuriiurshuliak** [comment]
The bug has been reproduced. To reproduce it, attempt to download all regions of Norway simultaneously. Once the download completes, you should observe that the Oslo map is missing.

OsmAnd~ 4.8.0#2533m, released: 2024-06-19

<table style="width:100%">
  </tr>
  </tr>
  <tr>
    <td> <video source src="https://github.com/osmandapp/OsmAnd/assets/127092082/d42fc854-5fd6-4ee4-b951-f47c5e9a7d9e"  width="300" height="auto"/>   </td>
    <td><video source src="https://github.com/osmandapp/OsmAnd/assets/127092082/c2a525d2-d3ed-4f85-ab79-05ea54b92f21"  width="300" height="auto"/>   </td>
  </tr>
</table>

**Comment by scaidermern** [comment]
Then I'm mistaken and Viken doesn't contain Oslo. This isn't obvious since when tapping on the Viken region on the map, OsmAnd displays it in such a way that Oslo is covered.

**Comment by vshcherb** [comment]
Probably bug is in Android, though if it's hard to fix we could extend map to contain Oslo similar to Brandenburg @Chumva @xmd5a2

**Comment by Chumva** [comment]
We filter some regions [here](https://github.com/osmandapp/Osmand/blob/66c4a8928125f548de9dc2ba84e7d3b45241566c/OsmAnd-java/src/main/java/net/osmand/map/WorldRegion.java#L255) by excluding nested subregions, avoiding duplication of what's already included in the top-level region

**Comment by xmd5a2** [comment]
If we actualize Norway subdivisions then issue will be fixed automatically. Oslo was inside Viken region which was short-lived: https://en.wikipedia.org/wiki/Viken_(county)

**Comment by vshcherb** [comment]
Yes we should follow new structure - 1 Viken

**Comment by xmd5a2** [comment]
New Norway regions was added: **Akershus, Buskerud, Finnmark, Østfold, Troms, Vestfold, Telemark**
Tifheightmap and SRTM were generated and uploaded. Regular maps will be generated at the beginning of September.
Old regions to remove: **troms-og-finnmark, viken**.

**Comment by DmitryAlexei** [comment]
Oslo now can be found as a separate map. 
Oslo and new regions Akershus, Buskerud, Finnmark, Østfold, Troms, Vestfold, Telemark, as well as the Norway contour lines and hillshades maps can be downloaded and work properly. 

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/db47da74-bee3-4c91-8e39-89bd65bef979"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/1cfd907d-f334-4006-82f7-9f058e1be550"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by pedrocvicente** [comment]
Hello!
Apologies for touching on an old topic, but I just updated my OSMAN app (on Android!) and in "maps & resources" it only lists 13 counties. One being Viken, although it doesn't exist anymore, and worse, it doesn't offer the option to download the standard map, only contour lines and terrain map. The only option to have that map seems to be to download the whole of Norway, which, given the size of the country, takes up quite some space. 
I see from the discussions that this should be fixed, but could it be its only fixed for iPhone but not android? 
Does anyone have any idea if this is going to be updated also from ANDroid??
Thanks!

-------
Brand : samsung
Manufacturer : samsung
Model : SM-G990B
Product : r9qxeea
Build : SP1A.210812.016.G990BXXS2CVI5
Version : 12
App Version : OsmAnd+
Apk Version : 4.8.6 4806

**Comment by vshcherb** [comment]
Did you check latest beta? Also check Other maps for 4.8.6 version

**Comment by bartgrefte** [comment]
Not sure if related, but after downloading all Norway-maps from download.osmand.net/list.php and deleting the years old (all in one) Norway map, I got an incomplete map of Norway. Last time I updated it was when Norway was just one big file, now a bunch of smaller ones.

After copying all the smaller regional maps to my phone (S20 FE 5G), not all of Norway appears green (as in map available). It's the area around Oslo (Buskerud, Akershus, Ostfold), but not Oslo itself which is available, and the regions above Sweden (Troms) and Finland (Finnmark).

Didn't ran into this with the Netherlands, also last updated when it was one big file, switching to the smaller files didn't result in any problems.
